napari/_qt/containers/qt_axis_model.py

```diff
@@ -71,8 +75,41 @@ def setData(
     ) -> bool:
         axis = self.getItem(index)
         if role == Qt.ItemDataRole.CheckStateRole:
-            axis.set_rollable(Qt.CheckState(value) == Qt.CheckState.Checked)
+            axis.rollable = Qt.CheckState(value) == Qt.CheckState.Checked
+        elif role == Qt.ItemDataRole.EditRole:
+            axis_labels = list(axis.dims.axis_labels)
+            axis_labels[axis.axis] = value
+            axis.dims.axis_labels = axis_labels
         else:
             return super().setData(index, value, role=role)
         self.dataChanged.emit(index, index, [role])
         return True
+
+    def flags(self, index: QModelIndex) -> Qt.ItemFlags:
+            """Returns the item flags for the given `index`.
+
+            This describes the properties of a given item in the model.  We set
+            them to be editable, checkable, dragable, droppable, etc...
+            If index is not a list, we additionally set `Qt.ItemNeverHasChildren`
+            (for optimization). Editable models must return a value containing
+            `Qt.ItemIsEditable`.
+
+            See Qt.ItemFlags https://doc.qt.io/qt-5/qt.html#ItemFlag-enum
+            """
+            flags = (
+                Qt.ItemFlag.ItemIsSelectable
```

Comment:
Should non-rollable dimensions be selectable?

I thought the main reason to select items in this list is to move multiple items. But maybe there are others?

Reply:
They have to be selectable otherwise you can't interact with them at all. ^^ Once unchecked their would be no chance to check them again.